osquery/tables/system/windows/shimcache.cpp

```diff
@@ -0,0 +1,216 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/core.h>
+#include <osquery/filesystem/fileops.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/tables/system/windows/userassist.h>
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+
+#include <string>
+
+const int WIN8 = 256;
+const int WIN10_PRE_CREATOR = 96;
+const int WIN10_CREATOR = 104;
+const std::string WIN8_START = "80";
+const std::string WIN10_START = "30";
+const std::string WIN10_CREATOR_START = "34";
+const std::string WIN_81_10_SHIMCACHE_DELIMITER = "31307473";
+
+// Shimcache can be in multiple ControlSets (ControlSet001, ControlSet002, etc)
+const std::string SHIMCACHE_CONTROLSET =
+    "HKEY_LOCAL_MACHINE\\SYSTEM\\%ControlSet%\\Control\\Session "
+    "Manager\\AppCompatCache";
+
+struct ShimcacheData {
+  std::string path;
+  long long last_modified;
+  std::string execution_flag;
+};
+
+namespace osquery {
+namespace tables {
+
+auto parseShimcacheData(std::string data,
```

Comment:
Pass strings by const reference to avoid making a copy of the data (unless you really want a copy of the data).

```
const std::string& data
```

repeat this for all the string parameters.

Reply:
Changed all string parameters